avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/RecordVisitor.java

```diff
@@ -31,6 +31,33 @@ public class RecordVisitor
      */
     protected final boolean _overridden;
 
+    /**
+     * When Avro schema for this JavaType ({@code _type}) results in UNION of multiple Avro types, _typeSchema keeps track
+     * which Avro type in the UNION represents this JavaType ({@code _type}) so that fields of this JavaType can be set to the right Avro type by {@code builtAvroSchema()}.
+     *
+     * Example:
+     * <pre>
+     *   @JsonSubTypes({
+     *     @JsonSubTypes.Type(value = Apple.class),
+     *     @JsonSubTypes.Type(value = Pear.class) })
+     *   class Fruit {}
+     *
+     *   class Apple extends Fruit {}
+     *   class Orange extends Fruit {}
+     * </pre>
+     * When _type = Fruit.class
+     * Then
+     * _avroSchema if Fruit.class is union of Fruit record, Apple record and Orange record schemas: [
+     *     { name: Fruit, type: record, field: [..] }, <--- _typeSchema points here
+     *     { name: Apple, type: record, field: [..] },
+     *     { name: Orange, type: record, field: [..]}
+     *   ]
+     * _typeSchema points to Fruit.class without subtypes record schema
+     *
+     * FIXME: When _thisSchema is not null, then _overridden must be true, therefore (_overridden == true) can be replaced with (_thisSchema != null),
```

Comment:
_thisSchema? Should it be _typeSchema? (or should _typeSchema be changed to _thisSchema)

Reply:
Right, fixed it. 

The comment is from when  the variable `_typeSchema` was called `_thisSchema`.

Reply:
Yeah so I figured.